airflowHPC/operators/resource_gmx_operator.py

```diff
@@ -70,43 +75,25 @@ def execute(self, context: Context):
                 raise ImportError(
                     "The gmx_executable argument must be set if the gmxapi package is not installed."
                 )
-        bash_path = shutil.which("bash") or "bash"
-        env = self.get_env(context)
-
-        self.log.info(f"mpi_executable: {self.mpi_executable}")
-        self.log.info(f"mpi_ranks: {self.mpi_ranks}")
-        self.log.info(f"gmx_executable: {self.gmx_executable}")
-        self.log.info(f"gmx_arguments: {self.gmx_arguments}")
-        self.log.info(f"input_files: {self.input_files}")
-        self.log.info(f"output_files: {output_files_paths}")
-        self.log.info(f"gpu_ids: {self.gpu_ids}")
-        self.log.info(f"hostname: {self.hostname}")
-
-        self.bash_command = self.create_gmxapi_call(
-            gmx_executable=self.gmx_executable,
-            gmx_arguments=self.gmx_arguments,
-            mpi_executable=self.mpi_executable,
-            mpi_ranks=self.mpi_ranks,
-            input_files=self.input_files,
-            output_files=output_files_paths,
-        )
-        result = self.subprocess_hook.run_command(
-            command=[bash_path, "-c", self.bash_command],
-            stdin=self.stdin,
-            env=env,
-            output_encoding=self.output_encoding,
-            cwd=self.output_dir,
-        )
-
-        if result.exit_code in self.skip_on_exit_code:
-            raise AirflowSkipException(
-                f"Bash command returned exit code {result.exit_code}. Skipping."
-            )
-        elif result.exit_code != 0:
-            raise AirflowException(
-                f"Bash command failed. The command returned a non-zero exit code {result.exit_code}."
-            )
-        return result.output
```

Comment:
The major issue I see here is the removal of the return statement. This return is necessary for any data to be stored from the task (technically this is not a hard requirement but it would be a PITA to work around), and also it is necessary to keep the worker executing the task busy, thereby preventing attempts to schedule further tasks.

Reply:
If I understand this correctly, result.output is the `stdout` stream of the executed process.  In RP we do not have that stream available - the process' `stdout` is always redirected to a file, and it is made available either via data staging or as a property on the task object, but only after the task completed.  What this basically means I think is that this `execute` operator is supposed to stay alive until the task completed and to then return the task's `stdout`?